src/tests/modules/bson.rs

```diff
@@ -345,3 +345,177 @@ fn debug_print() {
     assert_eq!(format!("{:?}", doc), normal_print);
     assert_eq!(format!("{:#?}", doc), pretty_print);
 }
+
+#[cfg(feature = "uuid-0_8")]
+#[test]
+fn test_binary_constructors() {
+    use crate::bson::UuidRepresentation;
+    use uuid::Uuid;
+
+    let uuid = Uuid::parse_str("00112233445566778899AABBCCDDEEFF").unwrap();
+    let bin = Binary::from_uuid(uuid);
+    assert_eq!(bin.bytes, uuid.as_bytes());
+    assert_eq!(bin.subtype, BinarySubtype::Uuid);
+
+    let bin = Binary::from_uuid_with_representation(uuid, UuidRepresentation::JavaLegacy);
+    assert_eq!(
+        bin.bytes,
+        Uuid::parse_str("7766554433221100FFEEDDCCBBAA9988")
+            .unwrap()
+            .as_bytes()
+    );
+    assert_eq!(bin.subtype, BinarySubtype::UuidOld);
+
+    let bin = Binary::from_uuid_with_representation(uuid, UuidRepresentation::CSharpLegacy);
+    assert_eq!(
+        bin.bytes,
+        Uuid::parse_str("33221100554477668899AABBCCDDEEFF")
+            .unwrap()
+            .as_bytes()
+    );
+    assert_eq!(bin.subtype, BinarySubtype::UuidOld);
+
+    let bin = Binary::from_uuid_with_representation(uuid, UuidRepresentation::PythonLegacy);
+    assert_eq!(
+        bin.bytes,
+        Uuid::parse_str("00112233445566778899AABBCCDDEEFF")
+            .unwrap()
+            .as_bytes()
+    );
+    assert_eq!(bin.subtype, BinarySubtype::UuidOld);
+}
+
+#[cfg(feature = "uuid-0_8")]
+#[test]
+fn test_binary_to_uuid() {
+    use crate::{bson::UuidRepresentation, de::Error};
+    use uuid::Uuid;
+
+    let uuid = Uuid::parse_str("00112233445566778899AABBCCDDEEFF").unwrap();
+    let bin = Binary::from_uuid(uuid);
+    // Error message for attempting to deserialize binary to a UUID using a non-standard
+    // representation when the representation is standard.
+    let error_msg = "expected binary subtype 3 when converting to UUID with a non-standard \
```

Comment:
these error messages can change from time to time, so it's best if we don't encode them in the tests. Otherwise we'll need to update this test each time we want to change the message. Instead, we can just verify that some error is returned. Ditto for the other tests.